### PR TITLE
[DMs] Fix pagination for chat list and chat message list

### PR DESCRIPTION
**Pull request by rickyrombo**
### Description

Fixes pagination for chat list and chat message list so that they can infinitely load.

Adds a new component, `StickyScrollList`, to help achieve the scrolling behavior desired for the `ChatMessagesList` component. It keeps the user scrolled to the bottom if they're at the bottom, and doesn't let them get stuck to the top when loading history.

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

*Please describe the tests that you ran to verify your changes. Provide repro instructions & any configuration.*

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/web/src/pages/chat-page/components/StickyScrollList.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import {
+  ComponentPropsWithoutRef,
+  forwardRef,
+  useLayoutEffect,
+  useRef
+} from 'react'
+
+import { mergeRefs } from 'react-merge-refs'
+
+type StickyScrollListProps = ComponentPropsWithoutRef<'div'> & {
```
Comment (rickyrombo) [inline packages/web/src/pages/chat-page/components/StickyScrollList.tsx:10]: @piazzatron look what i made! 😁

Comment (piazzatron) [reply packages/web/src/pages/chat-page/components/StickyScrollList.tsx:10]: ooo

Comment (piazzatron) [reply packages/web/src/pages/chat-page/components/StickyScrollList.tsx:10]: this is great

File: packages/web/src/pages/chat-page/components/StickyScrollList.tsx [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  useLayoutEffect(() => {
+    if (ref.current) {
+      if (resetKey !== oldResetKey.current) {
+        oldResetKey.current = resetKey
+
+        // If re-initting something that's already loaded
+        // this will make us start from the bottom again after render
+        ref.current.scrollTo({ top: ref.current.scrollHeight })
```
Comment (rickyrombo) [inline packages/web/src/pages/chat-page/components/StickyScrollList.tsx:87]: It's a little biased towards my use case, but easily generalizable if need be

**Comment by audius-infra** [comment]
Preview this change https://demo.audius.co/mjp-dms-pagination

File: packages/common/src/store/pages/chat/slice.ts [hunk 3]
```diff
@@ -132,6 +128,41 @@ const slice = createSlice({
...
+    },
+    updateMessageReactions: (
+      state,
+      action: PayloadAction<{ chatId: string; message: ChatMessage }>
+    ) => {
+      const { chatId, message } = action.payload
+      const index = state.chatMessages[chatId].data.findIndex(
+        (m) => m.message_id === message.message_id
+      )
+      state.chatMessages[chatId].data[index].reactions = message.reactions
```
Comment (sddioulde) [inline packages/common/src/store/pages/chat/slice.ts:165]: guessing we're sure that index > -1 here?

File: packages/web/src/pages/chat-page/components/ChatList.tsx [hunk 4]
```diff
@@ -42,24 +49,35 @@ export const ChatList = (props: ChatListProps) => {
...
+          status === Status.IDLE ||
+          (summary?.prev_count !== undefined && summary?.prev_count > 0)
+        }
+        useWindow={false}
+        loader={
+          <LoadingSpinner key={'loading-spinner'} className={styles.spinner} />
+        }
+      >
+        {chats?.length > 0 ? (
+          chats?.map((chat) => (
```
Comment (sddioulde) [inline packages/web/src/pages/chat-page/components/ChatList.tsx:66]: nit: don't need ? here

**Review by sddioulde — APPROVED** [review]
lgtm.

File: packages/common/src/store/pages/chat/slice.ts (hunk 3 above)
Comment (rickyrombo) [reply packages/common/src/store/pages/chat/slice.ts:165]: good call, will add some safeties

**Pull request merged**